modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -3908,6 +3908,12 @@ public CodegenResponse fromResponse(String responseCode, ApiResponse response) {
                     throw new RuntimeException("Invalid response code " + responseCode);
             }
         }
+
+        // for code ending with XX, e.g. 4XX, 5XX
+        if (r.code.length() == 3 && "XX".equals(r.code.substring(1))) {
+            r.wildcardFirstChar = r.code.substring(0, 1);
```

Comment:
It seems like rather than being a settable field, this property on response objects can be a calculated getter. See my comment in that class.